packages/@aws-cdk/cloud-assembly-schema/lib/assembly-manifest.ts

```diff
@@ -0,0 +1,340 @@
+interface BaseAssetMetadataEntry {
+    /**
+     * Requested packaging style
+     */
+    readonly packaging: string;
+
+    /**
+     * Logical identifier for the asset
+     */
+    readonly id: string;
+
+    /**
+     * The hash of the source directory used to build the asset.
+     */
+    readonly sourceHash: string;
+
+    /**
+     * Path on disk to the asset
+     */
+    readonly path: string;
+
+}
+
+/**
+ * Metadata Entry spec for files.
+ */
+export interface FileAssetMetadataEntry extends BaseAssetMetadataEntry {
+    /**
+     * Requested packaging style
+     */
+    readonly packaging: 'zip' | 'file';
+
+    /**
+     * Name of parameter where S3 bucket should be passed in
+     */
```

Comment:
Same here as the previous comment about clearer documentation.

Reply:
see https://github.com/aws/aws-cdk/pull/6724#discussion_r394291164